install.ps1

```diff
@@ -75,6 +75,39 @@ param (
 )
 $ErrorActionPreference = "Stop"
 
+# Function to test the network stack. Ping/GET requests to the resource to ensure that network stack looks good for installation
+function Test-WebConnection {
+    param (
+        [string]$url
+    )
+
+    Write-Host "[+] Checking for Internet connectivity ($url)..."
+
+    if (-not (Test-Connection $url -Quiet)) {
+        Write-Host "`t[!] It looks like you cannot ping $url. Check your network settings." -ForegroundColor Red
+        Start-Sleep 3
+        exit 1
+    }
+
+    $response = $null
+    try {
+        $response = Invoke-WebRequest -Uri "https://$url" -UseBasicParsing
+    }
+    catch {
+        Write-Host "`t[!] Error accessing $url. Exception: $($_.Exception.Message)`n`t[!] Check your network settings." -ForegroundColor Red
+        Start-Sleep 3
+        exit 1
+    }
+
+    if ($response -and $response.StatusCode -ne 200) {
+        Write-Host "`t[!] Unable to access $url. Status code: $($response.StatusCode)`n`t[!] Check your network settings." -ForegroundColor Red
+        Start-Sleep 3
+        exit 1
+    }
+
+    Write-Host "`t[+] Web check for $url passed" -ForegroundColor Green
```

Comment:
[nitpick] Should we remove the `Web` here?

Reply:
"Internet connectivity check passed" work here?